diff --git a/Makefile b/Makefile
new file mode 100644
index 0000000..36432ba
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,3 @@
+all:
+	chmod +x build_actions.sh
+	./build_actions.sh
\ No newline at end of file
diff --git a/build_actions.sh b/build_actions.sh
new file mode 100755
index 0000000..ba080e6
--- /dev/null
+++ b/build_actions.sh
@@ -0,0 +1,21 @@
+case "$TRAVIS_OS_NAME" in
+  osx)
+	brew install qt@5
+	export PATH="/usr/local/opt/qt@5/bin:$PATH"
+	export LDFLAGS="-L/usr/local/opt/qt@5/lib"
+	export CPPFLAGS="-I/usr/local/opt/qt@5/include"
+	export PKG_CONFIG_PATH="/usr/local/opt/qt@5/lib/pkgconfig"
+	;;
+  linux)
+	sudo apt-get install -y -qq qt5-default
+	;;
+  windows)
+	export QMAKE_PREFIX=/mingw64/qt5-static/bin/
+	pacman --needed --noconfirm -S mingw-w64-x86_64-qt5-static
+	;;
+esac
+
+"$QMAKE_PREFIX"qmake -o Makefile_qmake
+make -f Makefile_qmake
+cp release/ddb_gui_qt5.* . ; true
+cp ddb_gui_qt5.dylib ddb_gui_qt5.so ; true
